src/App.tsx

```diff
@@ -16,188 +229,151 @@ export const App: React.FC = () => {
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">
-              Todo is being saved now
-            </span>
+              {!(selectedTodo && selectedTodo.id === todo.id) ? (
+                <>
+                  <span
+                    data-cy="TodoTitle"
+                    className="todo__title"
+                    onDoubleClick={() => handleEdit(todo)}
+                  >
+                    {todo.title}
+                  </span>
 
-            <button type="button" className="todo__remove" data-cy="TodoDelete">
-              ×
-            </button>
+                  {/* Remove button appears only on hover */}
+                  <button
+                    type="button"
+                    className="todo__remove"
+                    data-cy="TodoDelete"
+                    onClick={() => deleteTodo(todo.id)}
+                  >
+                    ×
+                  </button>
+                </>
+              ) : (
+                <form onSubmit={() => handleUpdateTitle(todo)}>
+                  <input
+                    data-cy="TodoTitleField"
+                    type="text"
+                    className="todo__title-field"
+                    placeholder="Empty todo will be deleted"
+                    value={editTitle}
+                    onChange={e => setEditTitle(e.target.value)}
+                    onBlur={() => defaultValue()}
+                    ref={inputRef}
+                  />
+                </form>
+              )}
 
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div data-cy="TodoLoader" className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
+              {/* overlay will cover the todo while it is being deleted or updated */}
+              <div
+                data-cy="TodoLoader"
+                className={cn('modal overlay', { 'is-active': loading })}
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
             </div>
-          </div>
+          ))}
         </section>
 
         {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer" data-cy="Footer">
-          <span className="todo-count" data-cy="TodosCounter">
-            3 items left
-          </span>
-
-          {/* Active link should have the 'selected' class */}
-          <nav className="filter" data-cy="Filter">
-            <a
-              href="#/"
-              className="filter__link selected"
-              data-cy="FilterLinkAll"
-            >
-              All
-            </a>
+        {checkFooterActive() && (
```

Comment:
Here you just need to check if there is at least on todo exists.
```suggestion
        {todos.length && (
```